src/pages/fi-pv-expenses/forms/PaymentVoucherExpensesForm.js

```diff
@@ -0,0 +1,741 @@
+import CustomTextArea from 'src/components/Inputs/CustomTextArea'
+import { useDocumentType } from 'src/hooks/documentReferenceBehaviors'
+import { formatDateFromApi, formatDateToApi } from 'src/lib/date-helper'
+import WorkFlow from 'src/components/Shared/WorkFlow'
+import { useWindow } from 'src/windows'
+import ExpensesCostCenters from 'src/components/Shared/ExpensesCostCenters'
+
+export default function FiPaymentVoucherExpensesForm({ labels, maxAccess: access, recordId, plantId }) {
+  const [subtotalSum, setSubtotalSum] = useState(0);
+  const [vatSum, setVatSum] = useState(0);
+  const [amountSum, setAmountSum] = useState(0);
+
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const { stack } = useWindow()
+
+  const { documentType, maxAccess, changeDT } = useDocumentType({
+    functionId: SystemFunction.PaymentVoucher,
+    access: access,
+    enabled: !recordId
+  })
+
+  const initialValues =  {
+    recordId: null,
+    reference: '',
+    accountId: '',
+    accountType: 3,
+    currencyId: null,
+    paymentMethod: '',
+    date: new Date(),
+    glId: null, 
+    amount: null,
+    checkNo: '',
+    checkbookId: null,
+    decimals: null,
+    notes: '',
+    exRate: 1,
+    rateCalcMethod: 1,
+    baseAmount: null, 
+    cashAccountId: null,
+    dtId: documentType?.dtId,
+    status: 1,
+    releaseStatus: null,
+    plantId: plantId,
+    contactId: null,
+    isVerified: false,
+    expenses: [
+      {
+        id: 1,
+        pvId: recordId || 0,
+        seqNo: 1,
+        etId: '',
+        subtotal: null,
+        vatAmount: null,
+        amount: null,
+        supplierName: '',
+        taxRef: '',
+        notes: '',
+        isVAT: false,
+        hasCostCenters: true,
+        costCenters: []
+      },  
+    ],
+  }
+
+  const invalidate = useInvalidate({
+    endpointId: FinancialRepository.PaymentVouchers.page
+  })
+
+
+  const { formik } = useForm({
+    initialValues,
+    maxAccess,
+    enableReinitialize: false,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      accountType: yup.string().required(),
+      currencyId: yup.number().required(),
+      date: yup.string().required(),
+      paymentMethod: yup.string().required(),
+      cashAccountId: yup.string().required(),
+      expenses: yup
+          .array()
+          .of(
+            yup.object().shape({
+              etId: yup.number().required(),
+              amount: yup.number().required(),
+            })
+          )
+          .required()
+    }),
+    onSubmit: async obj => {
+      const copy = { ...obj }
+      delete copy.expenses
+      copy.date = formatDateToApi(copy.date)
+      copy.amount = totalAmount
+      copy.baseAmount = totalAmount
+      const costCenters = [];
+
+      const updatedRows = formik.values.expenses.map((expensesDetails, index) => {
+        const seqNo = index + 1
```

Comment:
try to destructured costCenters directly from expensesDetails
 const { costCenters, ...restDetails } = expensesDetails;
  if (costCenters) {
    costCenters.push(...costCenters);
  }